File: dongwoo/week3/hard.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int x = 0;
        int y = 0;

        for (int l : lottos) {
            if (l == 0) x++;
        }

        for (int i = 0; i < win_nums.length; i++) {
            for (int l : lottos) {
                if (l == win_nums[i]) {
                    y++;
                    break;
                }
            }
        }

        int happy = x + y;

        if (happy  < 2) happy = 1;
        if (y == 0) y = 1;

        return new int[] { 7 - happy, 7 - y };
    }
}

File: seunggon/week3/hard.py
def solution(lottos, win_nums):
    answer = [0,0]
    
    # 5위까지는 순위 + 일치하는 개수 = 7 성립
    # 1개일치 or 0개 일치 = 6위
    
    matchCnt = 0
    zeroCnt = 0

    for i in lottos:
        if i == 0:
            zeroCnt +=1
        elif i in win_nums:
            matchCnt +=1
    
    answer[0] = 6 if 7-zeroCnt-matchCnt > 6 else 7-zeroCnt-matchCnt 
    answer[1] = 6 if 7-matchCnt > 6 else 7-matchCnt
    
            
    
    return answer